packages/devextreme/js/__internal/core/state_manager/redux_dev_tools_connector.ts

```diff
@@ -0,0 +1,143 @@
+/* eslint-disable spellcheck/spell-checker */
+import { EventEmitter } from './event_emitter';
+import type * as StateManagementTypes from './types';
+
+export class ReduxDevToolsConnector implements StateManagementTypes.DevToolsConnector {
+  private devTools: StateManagementTypes.ReduxDevToolsInstance | null = null;
+
+  private isConnected = false;
+
+  private readonly logger: StateManagementTypes.Logger;
+
+  private readonly componentName: string;
+
+  private readonly externalActionEmitter:
+  EventEmitter<StateManagementTypes.DevToolsExternalActionCallback>;
+
+  constructor(componentName: string, logger: StateManagementTypes.Logger) {
+    this.componentName = componentName;
+    this.logger = logger;
+    this.externalActionEmitter = new EventEmitter(
+      'externalAction',
+      logger,
+    );
+  }
+
+  connect(): void {
+    try {
+      if (this.hasReduxDevTools(window)) {
+        this.devTools = window.__REDUX_DEVTOOLS_EXTENSION__.connect({
+          name: `${this.componentName} ${new Date().valueOf()}`,
+          trace: true,
+          traceLimit: 25,
+          features: {
+            jump: true,
+            skip: false,
+            dispatch: true,
+          },
+          shouldCatchErrors: true,
+          serialize: {
+            options: {
+              circular: '[CIRCULAR]',
+              date: true,
+            },
+            replacer: (key, value) => {
+              // replaced because this property contains a reference to the component instance
+              // which causes "heap out of memory"
+              if (key === 'changes' && (value !== undefined && value !== null) && typeof value === 'object' && 'component' in value && 'element' in value) {
+                return '[REPLACED]';
+              }
+
+              return value;
+            },
+          },
+        });
+
+        this.devTools.subscribe(
+          (message) => {
+            if (message.type === 'DISPATCH') {
+              if (message.payload.type === 'JUMP_TO_STATE' || message.payload.type === 'JUMP_TO_ACTION') {
+                if (message.state) {
+                  this.handleJumpToAction(message.state);
+                }
+              } else if (message.payload.type === 'COMMIT') {
+                this.externalActionEmitter.emit('COMMIT', null);
+              } else if (message.payload.type === 'RESET') {
+                this.externalActionEmitter.emit('RESET', null);
+              }
+            }
+          },
+        );
```

Comment:
Let's reduce the `connect` method complexity

We can reduce nesting here:
```suggestion
        this.devTools.subscribe(
          ({ type, payload: { type: payloadType }, state }) => {
            if (type !== 'DISPATCH') {
              return;
            }
            
            switch (payloadType) {
              case 'JUMP_TO_STATE':
              case 'JUMP_TO_ACTION':
                this.handleJumpToAction(state);
                break;
              case 'COMMIT':
                this.externalActionEmitter.emit('COMMIT', null);
                break;
               case 'RESET':
                 this.externalActionEmitter.emit('RESET', null);
                 break;
               default:
                 break;
            }
        });
```

Reply:
Changed